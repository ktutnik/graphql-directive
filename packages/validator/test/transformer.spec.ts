import val from "../src"
import { makeExecutableSchema } from "@graphql-tools/schema"
import { graphql } from "graphql"


it("Should execute validator on field argument", async () => {
    const typeDefs = /* GraphQL */ `
        type Query { name:String! }
        input User {
            name: String!
            email: String! @validate(method: EMAIL)
            spouse: User
        }
        
        type Mutation { 
            addUser(user:User!):String!
        }
    `
    const schema = val.transform(makeExecutableSchema({
        typeDefs: [val.typeDefs, typeDefs], 
        resolvers: {
            Mutation: {
                addUser: (_, args) => {
                    return "Success"
                }
            }
        } 
    }))
    const result = await graphql({ schema, source: `mutation { addUser(user: { name: "John", email: "mail" }) }`})
    expect(result.errors![0].extensions).toMatchSnapshot()
    expect(result.data).toBe(10)
})