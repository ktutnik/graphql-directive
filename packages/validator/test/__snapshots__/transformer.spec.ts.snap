// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Type Validation Should validate argument with primitive type on Custom Type 1`] = `
{
  "error": [
    {
      "message": "Must be a valid email address",
      "path": "email.checkEmail.email",
    },
  ],
}
`;

exports[`Custom Type Validation Should validate argument with primitive type on Custom Type 2`] = `
{
  "email": {
    "checkEmail": true,
  },
}
`;

exports[`Mutation Validation Should validate argument inside custom type property 1`] = `
{
  "error": [
    {
      "message": "Must be a valid email address",
      "path": "addUser.user.email",
    },
  ],
}
`;

exports[`Mutation Validation Should validate argument inside custom type property 2`] = `
{
  "addUser": true,
}
`;

exports[`Mutation Validation Should validate argument inside nested custom type property 1`] = `
{
  "error": [
    {
      "message": "Must be a valid email address",
      "path": "addUser.user.suppose.email",
    },
  ],
}
`;

exports[`Mutation Validation Should validate argument inside nested custom type property 2`] = `
{
  "addUser": true,
}
`;

exports[`Mutation Validation Should validate argument with primitive type properly 1`] = `
{
  "error": [
    {
      "message": "Must be a valid email address",
      "path": "checkEmail.email",
    },
  ],
}
`;

exports[`Mutation Validation Should validate argument with primitive type properly 2`] = `
{
  "checkEmail": true,
}
`;
